### PR TITLE
[script] [crossing-repair] Accept any town arg that `DRC.get_town_name` supports

### PR DESCRIPTION
**Pull request by KatoakDR**
### Dependencies
* https://github.com/rpherbig/dr-scripts/pull/5323
* https://github.com/rpherbig/dr-scripts/pull/5345

### Background
* `crossing-repair` supports a town argument to override your hometown in your yaml to specify where to repair your gear.
* However, it only supports Crossing, Riverhaven, Theren, Shard, Hibarnhvidar, and Ratha as town overrides even though other towns support repairs, like Fang Cove.

### Changes
* Remove restriction on town name in the script argument regex pattern
* Replace hard-coded if/else parsing of town name with `DRC.get_town_name(..)`
* Error if town argument is not recognized
* Remove unused variable `@hometown`

### Fang Cove
_My hometown is Crossing but repaired in Fang Cove_
```
>,crossing-repair fang
...
(went to fang cove and repaired all my gear)
```

### Unrecognized town
```
>,crossing-repair foo

--- Lich: crossing-repair active.

| Could not identify town for argument: foo
| To avoid ambiguity, please use the town's full name: https://elanthipedia.play.net/Category:Cities

--- Lich: crossing-repair has exited.
```

### CODE REVIEW

File: crossing-repair.lic [hunk 1]
```diff
@@ -178,7 +178,7 @@ class CrossingRepair
   def setup
     arg_definitions = [
       [
-        { name: 'town', regex: /Crossing|River.*|Theren.*|Shard|Hib.*|Ratha/i, optional: true, description: 'Town to sell in' },
+        { name: 'town', regex: /^(?!force|debug).+$/, optional: true, description: 'Town to sell in' },
```
Comment (rpherbig) [inline crossing-repair.lic:181]: What do you think about using `DRC.$HOMETOWN_REGEX` (or maybe you can just use `$HOMETOWN_REGEX`?) here?

Comment (KatoakDR) [reply crossing-repair.lic:181]: I'll try that out, thanks

Comment (KatoakDR) [reply crossing-repair.lic:181]:
```suggestion
        { name: 'town', regex: $HOMETOWN_REGEX, optional: true, description: 'Town to sell in' },
```

Comment (KatoakDR) [reply crossing-repair.lic]: This change depends on https://github.com/rpherbig/dr-scripts/pull/5345

**Comment by KatoakDR** [comment]
Dependencies are merged, waiting another day or so for those to soak in before mark this ready for review

**Comment by KatoakDR** [comment]
@rpherbig this is ready for your review/merge, thanks!

**Pull request merged**